src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -229,11 +248,10 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
 
   useEffect(() => {
     const unsubscribe = txSubscribe(TxEvent.SAFE_APPS_REQUEST, async ({ safeAppRequestId, safeTxHash }) => {
-      const currentSafeAppRequestId = currentRequestId
-
-      if (currentSafeAppRequestId === safeAppRequestId) {
+      if (safeAppRequestId && currentRequestId === safeAppRequestId) {
```

Comment:
I've refactored this to be consistent with the same logic that we use for signing messages in line 263